### PR TITLE
feat: create api test for safaricom AB#28905

### PR DESCRIPTION
**Pull request by Vladimir-Petkov**
## Describe your changes

Brief description of your changes - not in-depth because the bulk of the description should be in the task on DevOps.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/test/payment/do-payment-safaricom.test.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        programId,
+        payment,
+        amount,
+        paymentReferenceIds,
+        accessToken,
+      );
+
+      // Assert
+      let getTransactionsBody: any[] = [];
+      while (getTransactionsBody.length <= 0) {
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/test/payment/do-payment-safaricom.test.ts]: done

File: services/121-service/test/payment/do-payment-safaricom.test.ts [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/test/payment/do-payment-safaricom.test.ts]: done

File: services/121-service/test/payment/do-payment-safaricom.test.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        paymentReferenceIds,
+        accessToken,
+      );
+
+      await waitForPaymentTransactionsToComplete(
+        programId,
+        paymentReferenceIds,
+        accessToken,
+        3001,
+        Object.values(StatusEnum),
```
Comment (RubenGeo) [inline services/121-service/test/payment/do-payment-safaricom.test.ts:169]: I think you should not include status 'waiting' here. Else the test could break if you get the transactions before the callback from the mock-service happens and that transaction is still on 'waiting'

**Review by RubenGeo — CHANGES_REQUESTED** [review]
see comment

File: services/121-service/test/payment/do-payment-safaricom.test.ts (hunk 3 above)
Comment (Vladimir-Petkov) [reply services/121-service/test/payment/do-payment-safaricom.test.ts]: done

**Review by RubenGeo — APPROVED** [review]

**Pull request merged**